tests/node_test.cc

```diff
@@ -249,4 +250,51 @@ TEST(NodeTest, MultipleSubscriber) {
     }
 }
 
+TEST(NodeTest, StrandSubscriber) {
+    static constexpr std::size_t kThreadNum       = 4;
+    constexpr std::size_t        kNumOfSubChannel = 4;
+    constexpr std::size_t        kMessageNumber   = 50000;
+
+    using TestMessageType = TestMessage<11>;
+
+    CRNode publisher;
+    CRNode subscriber;
+
+    auto runner = JobRunner::MakeJobRunner(JobRunner::Config{
+        .thread_num_ = kThreadNum,
+    });
+    subscriber.SetRunner(runner);
+
+    std::mutex                                cv_mutex;
+    std::condition_variable                   cv;
+    std::array<std::size_t, kNumOfSubChannel> channel_counter{0};
+    std::array<bool, kNumOfSubChannel>        complete{0};
+
+    for (std::size_t channel_subid = 0; channel_subid < kNumOfSubChannel; ++channel_subid) {
+        subscriber.Subscribe<TestMessageType>(
+            channel_subid,
+            [&counter = channel_counter[channel_subid], &complete = complete[channel_subid], &cv_mutex, &cv](
+                const std::shared_ptr<TestMessageType>& message) {
+                EXPECT_EQ(counter++, message->value_);
+                if (counter == kMessageNumber) {
+                    std::unique_lock cv_lock(cv_mutex);
+                    complete = true;
+                    cv.notify_all();
+                }
+            },
+            /* allow_concurrency = */ false);
+    }
+
+    for (std::size_t msg_idx = 0; msg_idx < kMessageNumber; ++msg_idx) {
+        for (std::size_t channel_subid = 0; channel_subid < kNumOfSubChannel; ++channel_subid) {
+            publisher.Publish(channel_subid, std::make_shared<TestMessageType>(msg_idx));
+        }
+    }
+
+    std::unique_lock cv_lock(cv_mutex);
+    cv.wait(cv_lock, [&complete]() {
+        return std::reduce(complete.begin(), complete.end(), true, std::logical_and<void>());
```

Comment:
include

Reply:
https://github.com/cyfitech/cris-core/pull/59/files/7ffc122a48987933934248128a30fb5f4faebab8#diff-d9095f2acba85a8dde7c76f62246bfd05541cedf36b8158a4495c809a93362c9R9

Reply:
logical_and is in functional